claragenomics/io/bedio.py

```diff
@@ -27,24 +28,27 @@ def read_intervals(bed_file):
     df = pd.read_csv(bed_file, sep='\t', header=None,
                      names=['chrom', 'start', 'end'],
                      usecols=(0, 1, 2),
-                     dtype={'chrom': str, 'start': int, 'end': int})
+                     dtype={'chrom': str, 'start': int, 'end': int},
+                     skiprows=skip)
     return df
 
 
 def read_sizes(sizes_file, as_intervals=False):
-    """Function to read chromosome sizes into a DataFrame.
+    """Read chromosome sizes into a DataFrame.
 
     Args:
         sizes_file(str): Path to sizes file
         as_intervals(bool): Format the DataFrame as 0-indexed intervals
 
+    Returns:
+        df: Pandas DataFrame
+
     """
     df = pd.read_csv(sizes_file, sep='\t', header=None, usecols=(0, 1),
-                     dtype={0: str, 1: int})
+                     names=['chrom', 'length'],
+                     dtype={'chrom': str, 'length': int})
     if as_intervals:
-        df[2] = [0] * len(df)
-        df.rename(columns={0: 0, 2: 1, 1: 2}, inplace=True)
-        df.columns = ['chrom', 'start', 'end']
-    else:
-        df.columns = ['chrom', 'length']
+        df['start'] = [0] * len(df)
+        df = df[['chrom', 'start', 'length']]
+        df.rename(columns={"length": "end"}, inplace=True)
```

Comment:
why is length being renamed to end ? May be leave a comment for this operation, so we are not confused later on.

Reply:
Added comments.

Reply:
It's done so that the output has the same format as the output of `read_intervals`. So that the same downstream function can take input either from `read_intervals` or `read_sizes`. Currently we use this in `calculate_baseline_metrics.py`.

Reply:
can we replace "length" with "end" in line 48. That way we don't have to rename it.

Reply:
Not straightforward. In both main.py and df_to_bedGraph, we have to merge the two dataframes (intervals and sizes) together. The merge won't work if they have the same column name ('end') so I would have to rename one of them anyway. Better to have different names from the beginning. Also in my opinion it's confusing to call a column 'end' in a dataframe where there is no column named 'start'.